src/Control/Distributed/Spark/PairRDD.hs

```diff
@@ -31,22 +33,26 @@ toRDD prdd = do
 
 fromRDD :: RDD (Tuple2 a b) -> IO (PairRDD a b)
 fromRDD rdd =
-  callStatic (sing :: Sing "org.apache.spark.api.java.JavaPairRDD")
-             "fromJavaRDD" [coerce rdd]
+  callStatic
+    "org.apache.spark.api.java.JavaPairRDD"
+    "fromJavaRDD"
+    [coerce rdd]
 
 join :: PairRDD a b -> PairRDD a c -> IO (PairRDD a (Tuple2 b c))
 join prdd0 prdd1 = call prdd0 "join" [coerce prdd1]
 
-keyBy :: Reflect (Closure (v -> k)) ty1
-      => Closure (v -> k) -> RDD v -> IO (PairRDD k v)
+keyBy
+  :: (Static (Reify v), Static (Reflect k), Typeable v, Typeable k)
```

Comment:
Is this `ReflectFun 1 (v -> k)`? It is better to use the most specific constraint that the function needs. Otherwise, the type signature will need to be changed every time the constraints of `ReflectFun 1 (v -> k)` change.

The same observation applies to the other functions which reflect functions.

Reply:
The two are isomorphic, rather than one being more specific than the other. So it doesn't change forward compatibility either way. There is the question of which one we'd want to show in the Haddocks to the user. I'm of two minds regarding which one is more edifying. Thoughts?

Reply:
I find more useful in documentation saying that the function needs to reflect the closure. It also saves duplicating the list of constraints of `ReflectFun 1 (a -> b)` at every function that uses it.

Reply:
Discussed offline but FTR: the above constraints are actually not equivalent to `ReflectFun 1 (a -> b)`, because they are not superclass constraints of `ReflectFun`. It was decided to therefore keep the constraints as-is.

Reply:
Or rather, they may well be equivalent in some sense, but GHC currently does not simplify one into the other, presumably because doing so would be incorrect if any instance anywhere was overlapping (they are not in this case, but GHC doesn't know that).